[Bug] gvisor handler function signature updated

### Verify steps

- [x] Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome.
- [x] I have searched on the [issue tracker](……/) for a related issue.

### Version

latest

### What OS are you seeing the problem on?

_No response_

### Description

```shell
#go get gvisor.dev/gvisor@go
go: downloading gvisor.dev/gvisor v0.0.0-20250807194038-c9af560a03d9
```

### CLI or Config

_No response_

### Logs

```shell
# github.com/xjasonlyu/tun2socks/v2/core
/Users/starx/go/pkg/mod/github.com/xjasonlyu/tun2socks/v2@v2.6.0/core/udp.go:16:39: cannot use func(r *udp.ForwarderRequest) {…} (value of type func(r *udp.ForwarderRequest)) as udp.ForwarderHandler value in argument to udp.NewForwarder
```

### How to Reproduce

_No response_

--- a/core/udp.go
+++ b/core/udp.go
@@ -13,7 +13,7 @@ import (
 
 func withUDPHandler(handle func(adapter.UDPConn)) option.Option {
 	return func(s *stack.Stack) error {
-		udpForwarder := udp.NewForwarder(s, func(r *udp.ForwarderRequest) {
+		udpForwarder := udp.NewForwarder(s, func(r *udp.ForwarderRequest) bool {
 			var (
 				wq waiter.Queue
 				id = r.ID()
@@ -22,7 +22,7 @@ func withUDPHandler(handle func(adapter.UDPConn)) option.Option {
 			if err != nil {
 				glog.Debugf("forward udp request: %s:%d->%s:%d: %s",
 					id.RemoteAddress, id.RemotePort, id.LocalAddress, id.LocalPort, err)
-				return
+				return false
 			}
 
 			conn := &udpConn{
@@ -30,6 +30,7 @@ func withUDPHandler(handle func(adapter.UDPConn)) option.Option {
 				id:      id,
 			}
 			handle(conn)
+			return true
 		})
 		s.SetTransportProtocolHandler(udp.ProtocolNumber, udpForwarder.HandlePacket)
 		return nil

--- a/go.mod
+++ b/go.mod
@@ -21,7 +21,7 @@ require (
 	golang.org/x/time v0.11.0
 	golang.zx2c4.com/wireguard v0.0.0-20250521234502-f333402bd9cb
 	gopkg.in/yaml.v3 v3.0.1
-	gvisor.dev/gvisor v0.0.0-20250523182742-eede7a881b20
+	gvisor.dev/gvisor v0.0.0-20250807194038-c9af560a03d9
 )
 
 require (

